src/main/java/com/gluonhq/richtext/RichTextAreaSkin.java

```diff
@@ -152,18 +164,36 @@ public void dispose() {
         getSkinnable().setEditable(false); // removes all related listeners
         getSkinnable().editableProperty().removeListener(this::editableChangeListener);
         viewModel.removeChangeListener(textChangeListener);
+        textBackgroundIndexRangeColors.removeListener(textBackgroundIndexRangeColorsChangeListener);
     }
 
     /// PRIVATE METHODS /////////////////////////////////////////////////////////
 
-    //TODO Need more optimal way of rendering text fragments.
+    // TODO Need more optimal way of rendering text fragments.
     //  For now rebuilding the whole text flow
     private void refreshTextFlow() {
         fontCacheEvictionTimer.pause();
         try {
             var fragments = new ArrayList<Text>();
-            viewModel.walkFragments((text, decoration) -> fragments.add(buildText(text, decoration)));
+            var bir = new ArrayList<IndexRangeColor>();
+            var length = new AtomicInteger();
+            viewModel.walkFragments((text, decoration) -> {
+                final Text textNode = buildText(text, decoration);
+                fragments.add(textNode);
+
+                if (decoration.getBackground() != Color.TRANSPARENT) {
+                    final IndexRangeColor indexRangeColor = new IndexRangeColor(
```

Comment:
hmm, this would be great to have, but doesn't work if you change for instance the font size of another piece
<img width="239" alt="image" src="https://user-images.githubusercontent.com/2043230/155488121-32c5930a-9fdd-4d63-a6c5-0c7d8aeff42f.png">

Reply:
I am going to revert the commit I made. Layoutbounds of a Path is calculated without adding it to the scene-graph. Performance impact should be minimal from my previous approach.